include/odp/api/spec/event.h

```diff
@@ -141,6 +142,27 @@ odp_event_type_t odp_event_types(odp_event_t event,
 int odp_event_type_multi(const odp_event_t event[], int num,
 			 odp_event_type_t *type);
 
+/**
+ * Filter and convert packet events
+ *
+ * Checks event type of all input events, converts all packet events and outputs
+ * packet handles. Returns the number packet handles outputted. Outputs the
+ * remaining, non-packet event handles to 'remain' array. Handles are outputted
+ * to both arrays in the same order those are stored in 'event' array. Both
+ * output arrays must fit 'num' elements.
+ *
+ * @param      event    Array of event handles
+ * @param[out] packet   Packet handle array for output
+ * @param[out] remain   Event handle array for output of remaining, non-packet
+ *                      events
+ * @param      num      Number of events (> 0)
+ *
+ * @return Number of packets outputted (0 ... num)
+ */
+int odp_event_filter_packet(const odp_event_t event[],
+			    odp_packet_t packet[],
+			    odp_event_t remain[], int num);
```

Comment:
Doesn't this reorder events? As described consider an array of events that consist of packets (p) and non-packets (np).

With an input array of events: np1, p1, np2, p2, p3 as described `odp_event_filter_packet()` would have RC = 3, the `packet[]` array would contain p1, p2, p3, and the `remain[]` event would contain np1, np2.  Is that behavior what we want?

Reply:
why not odp_schedule_type(ODP_PACKET,  void  *array) ? I.e. if you need only packets then do scheduling only for patches. That has to be more fast than get all and then filer them.

Reply:
" Handles are outputted to both arrays in the same order those are stored in 'event' array. "

Events are not reordered. Packets are copied into packet[] and non-packets into remain[]. Both arrays maintain p and np order of the original array. Also original array is not modified.

packet[] = p1, p2, p3
remain[] = np1, np2

Reply:
Schedulers do not work that way. You cannot peak into head of schedule queue and ignore non-packets and just select packets. Scheduler returns N events.

Also storing non-packets locally (inside scheduler SW) would not help, since application could easily starve non-packets by polling those too seldom.